src/puya/awst_build/module.py

```diff
@@ -151,8 +152,25 @@ def visit_class_def(self, cdef: mypy.nodes.ClassDef) -> None:
             # TODO: other checks above?
             else:
                 name_override: str | None = None
+                scratch_slot_reservations = list[ScratchSpaceReservation]()
                 for kw_name, kw_expr in cdef.keywords.items():
                     with self.context.log_exceptions(kw_expr):
+                        if kw_name == "scratch_slots":
```

Comment:
I think it'd be better to fold this into the match below

Reply:
We could, but `kw_value = kw_expression.accept(self)` throws an error for the scratch_slots parameter - unless we want to implement the relevant visitors so that just works?

Reply:
fixed by combining them